src/utils.ts

```diff
@@ -0,0 +1,99 @@
+export const timeStamp = (): string => {
+  const now = new Date();
+  return now.toLocaleString();
+};
+
+const isHTMLElement = (element: Element | null): element is HTMLElement => {
+  return element instanceof HTMLElement;
+};
+
+export const isInputElement = (activeElement: Element | null) => {
+  if (
+    activeElement &&
+    (activeElement.tagName === "INPUT" ||
+     activeElement.tagName === "TEXTAREA" || 
+     (isHTMLElement(activeElement) && activeElement.isContentEditable))
+  ) {
+    return true;
+  }
+};
+
+export const keyMap: Record<string, { shifted: string; unshifted: string }> = {
+  "ShiftLeft": { shifted: "Shift", unshifted: "Shift" },
+  "ShiftRight": { shifted: "Shift", unshifted: "Shift" },
+  "ControlLeft": { shifted: "Ctrl", unshifted: "Ctrl" },
+  "ControlRight": { shifted: "Ctrl", unshifted: "Ctrl" },
+  "AltLeft": { shifted: "Alt", unshifted: "Alt" },
+  "AltRight": { shifted: "Alt", unshifted: "Alt" },
+  "MetaLeft": { shifted: "Command", unshifted: "Command" },
+  "MetaRight": { shifted: "Command", unshifted: "Command" },
+  "KeyA": { shifted: "A", unshifted: "a" },
+  "KeyB": { shifted: "B", unshifted: "b" },
+  "KeyC": { shifted: "C", unshifted: "c" },
+  "KeyD": { shifted: "D", unshifted: "d" },
+  "KeyE": { shifted: "E", unshifted: "e" },
+  "KeyF": { shifted: "F", unshifted: "f" },
+  "KeyG": { shifted: "G", unshifted: "g" },
+  "KeyH": { shifted: "H", unshifted: "h" },
+  "KeyI": { shifted: "I", unshifted: "i" },
+  "KeyJ": { shifted: "J", unshifted: "j" },
+  "KeyK": { shifted: "K", unshifted: "k" },
+  "KeyL": { shifted: "L", unshifted: "l" },
+  "KeyM": { shifted: "M", unshifted: "m" },
+  "KeyN": { shifted: "N", unshifted: "n" },
+  "KeyO": { shifted: "O", unshifted: "o" },
+  "KeyP": { shifted: "P", unshifted: "p" },
+  "KeyQ": { shifted: "Q", unshifted: "q" },
+  "KeyR": { shifted: "R", unshifted: "r" },
+  "KeyS": { shifted: "S", unshifted: "s" },
+  "KeyT": { shifted: "T", unshifted: "t" },
+  "KeyU": { shifted: "U", unshifted: "u" },
+  "KeyV": { shifted: "V", unshifted: "v" },
+  "KeyW": { shifted: "W", unshifted: "w" },
+  "KeyX": { shifted: "X", unshifted: "x" },
+  "KeyY": { shifted: "Y", unshifted: "y" },
+  "KeyZ": { shifted: "Z", unshifted: "z" },
+  "Comma": { shifted: "<", unshifted: "," },
+  "Period": { shifted: ">", unshifted: "." },
+  "Slash": { shifted: "?", unshifted: "/" },
+  "Semicolon": { shifted: ":", unshifted: ";" },
+  "Quote": { shifted: "\"", unshifted: "'" },
+  "BracketLeft": { shifted: "{", unshifted: "[" },
+  "BracketRight": { shifted: "}", unshifted: "]" },
+  "Backslash": { shifted: "|", unshifted: "\\" },
+  "Backquote": { shifted: "~", unshifted: "`" },
+  "Minus": { shifted: "_", unshifted: "-" },
+  "Equal": { shifted: "+", unshifted: "=" }
+};
+
+export const isShortcutPressed = (pressedKeys: Set<string>, shortcutKeys: string): boolean => {
```

Comment:
I wonder if you could avoid some of this complexity by using a hotkeys library like CLUE does.

Reply:
@bgoldowsky Thanks. I did try some libraries previously, but the ones I tried didn't play nicely with the need to set the shortcut listener in both the plugin and in the main CODAP window. I think maybe the one CLUE uses may not have that problem, though. I'll give it a try.

Reply:
I tried out is-hotkey. It was promising in that it did work for both adding a listener to the plugin and the main CODAP window. I ran into another issue, though, with the shift key which I couldn't get around. I think there must be a solution to that, but it's not obvious to me at this point. I'm going to leave the existing shortcut-handling code for now. I do think using a library would be better, though, and will make sure we revisit the idea after the UI gets refined.